src/Streamly/Internal/LZ4/Config.hs

```diff
@@ -61,6 +74,12 @@ data Config a =
         , getUncompSize :: Ptr Word8 -> IO Int32
         , dataOffset :: Int
         , compSizeOffset :: Int
+        -- If checksum is enabled this will verify the sum and return
+        -- False. False tells the combinator not to run the default resizer but
+        -- to stop. Having "ResizeState" here would make this more general
+        -- though.
+        , resizeGuard :: Ptr Word8 -> IO ResizeGuard
```

Comment:
We can call this function as `readBlock` or `peekBlock`, we are essentially trying to read the next data block in the stream.